File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int res = 0;
        int buy = 0;
        for (int sell = 0; sell < prices.length; sell++) {
            if (prices[buy] > prices[sell]) {
                buy = sell;
            } else {
                res = Math.max(res, prices[sell]-prices[buy]);
            }
        }
        return res;
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int res = 0;
        int min2 = prices[0];
        for (int i = 0; i < n; i++) {
            min2 = min(min2, prices[i]);
            res = max(res, prices[i]-min2);
        }
        return res;
    }
    
};